lighthouse-core/gather/computed/trace-of-tab.js

```diff
@@ -45,18 +45,30 @@ class TraceOfTab extends ComputedArtifact {
       .filter(e => {
         return e.cat.includes('blink.user_timing') ||
           e.cat.includes('loading') ||
-          e.cat.includes('devtools.timeline') ||
-          e.name === 'TracingStartedInPage';
+          e.cat.includes('devtools.timeline');
       })
       // @ts-ignore - stableSort added to Array by WebInspector.
       .stableSort((event0, event1) => event0.ts - event1.ts);
 
-    // The first TracingStartedInPage in the trace is definitely our renderer thread of interest
-    // Beware: the tracingStartedInPage event can appear slightly after a navigationStart
-    const startedInPageEvt = keyEvents.find(e => e.name === 'TracingStartedInPage');
-    if (!startedInPageEvt) throw new LHError(LHError.errors.NO_TRACING_STARTED);
-    // @ts-ignore - property chain exists for 'TracingStartedInPage' event.
+    // Find out the inspected page frame.
+    let startedInPageEvt;
+    const startedInBrowserEvt = keyEvents.find(e => e.name === 'TracingStartedInBrowser');
+    if (startedInBrowserEvt && startedInBrowserEvt.args.data.persistentIds) {
+      const mainFrame = (startedInBrowserEvt.args.data.frames || []).find(frame => !frame.parent);
+      const mainFrameId = mainFrame && mainFrame.frame || '';
```

Comment:
what's the use of the fallback `''` value here if there's no `mainFrame` or `mainFrameId` found? Seems like we need those and should just throw if not found

(or could set `startedInPageEvt` to `undefined` and let the legacy fallback below try to find a backup `TracingStartedInPage` event)

Reply:
Agree